docs/cluster_spec.md

```diff
@@ -101,6 +101,45 @@ spec:
       type: Public
 ```
 
+### Load Balancer Subnet configuration 
+
+**AWS only**
+
+By default, kops will try to choose one suitable subnet per availability zone and use these for the API load balancer.
+Depending on the `type`, kops will choose from either `Private` or `Public` subnets. If this default logic is not
+suitable for you (e.g. because you have a more granular separation between subnets), you can explicitly configure
+the to-be-use subnets:
+
+```yaml
+spec:
+  api:
+    loadBalancer:
+      type: Public
+      subnets:
+        - name: subnet-a
+        - name: subnet-b
+        - name: subnet-c
+````
+
+It is only allowed to add more subnets and forbidden to remove existing ones. This is due to limitations on AWS
+ELBs and NLBs.
```

Comment:
Is there the possibility to remove existing ones if you accept that the consequence is destroying the loadbalancer and creating a new one? eg `--force-recreate-loadbalancer`
I'd consider losing access to the loadbalancer acceptible during a maintenance window.

Reply:
I'd assume that this would be possible. But this is not a new problem and thus should be part of another PR. Currently, when you change your set of shared subnets, you'd run into the same issue already as the automatically chosen subnets would differ then.

Reply:
My question was primarily around the validation side of things. If I wanted to change those values and accepted that I'd have to (manually) remove the loadbalancer,  would your changes to kops reject my edits?

Reply:
It would accept the edits and only fail when running the update. The failure is a little bit unfortunate as it keeps retrying but then of course never succeeds.

Reply:
A workaround for the above behavior is to manually delete the load balancer. A `kops update cluster` should then report that it will (re)create the load balancer which should succeed with `--yes`. You'll run into certificate issues with gossip clusters though since the load balancer's DNS name is used in various certs.

Reply:
Ah I see you mentioned this a few lines below. I believe you'll also need to `kops rolling-update cluster --instance-group-roles=Master --cloudonly` after `kops update cluster --yes` recreates the NLB in order to pick up new certificates.

Reply:
@rifelpet I did not observe the need for rolling-update while doing my testing, which I did on a semi-live (not prod, but a lot of stuff deployed) cluster.

Reply:
It may only be gossip clusters that need the rolling update (cluster names ending in `.k8s.local`, was yours gossip or using route53 ? Now that I think about it more, I think the nodes themselves might need to be rolled too to pick up the new load balancer dns name which is pretty invasive. We could say "It is not recommended to make this change on gossip clusters due to the requirement of performing a --cloudonly rolling update on the entire cluster which results in temporary downtime of all workloads. It is recommended to create a new cluster instead."